### PR TITLE
Use new filters format in the upload plugin

### PR DESCRIPTION
**Pull request by petersg83**

### CODE REVIEW

File: packages/core/upload/server/controllers/content-api.js [hunk 1]
```diff
@@ -118,4 +119,35 @@ module.exports = {
...
+  },
+
+  async search(ctx) {
+    const { id } = ctx.params;
+    const model = strapi.getModel('plugin::upload.file');
+    const entries = await strapi.query('plugin::upload.file').findMany({
+      where: {
+        $or: [{ hash: { $contains: id } }, { name: { $contains: id } }],
+      },
+    });
```
Comment (petersg83) [inline packages/core/upload/server/controllers/content-api.js:149]: put common code in a service

File: packages/core/upload/server/services/upload.js [hunk 2]
```diff
@@ -44,10 +44,11 @@ const sendMediaMetrics = data => {
 };
 
 const combineFilters = params => {
-  // FIXME: until we support boolean operators for querying we need to make mime_ncontains use AND instead of OR
   if (_.has(params, 'mime_ncontains') && Array.isArray(params.mime_ncontains)) {
-    params._where = params.mime_ncontains.map(val => ({ mime_ncontains: val }));
+    const mimeFilters = { $or: params.mime_ncontains.map(val => ({ mime_ncontains: val })) };
```
Comment (petersg83) [inline packages/core/upload/server/services/upload.js:48]: ```mime: { $notContainsi: val }```

Comment (petersg83) [reply packages/core/upload/server/services/upload.js:48]: use $and instead

Comment (petersg83) [reply packages/core/upload/server/services/upload.js:48]: Maybe remove this function, it's better if the front do it

File: packages/core/upload/server/services/upload.js [hunk 3]
```diff
@@ -311,18 +312,18 @@ module.exports = ({ strapi }) => ({
     return res;
   },
 
-  findOne(params, populate) {
-    return strapi.query('plugin::upload.file').findOne({ where: params, populate });
+  findOne(filters, populate) {
+    return strapi.entityService.findOne('plugin::upload.file', { params: { filters, populate } });
```
Comment (petersg83) [inline packages/core/upload/server/services/upload.js:316]: ```return strapi.entityService.findOne('plugin::upload.file', id, { params: { populate } });```
(no param with Alex's PR)

File: packages/core/admin/server/services/permission/permissions-manager/query-builers.js [hunk 4]
```diff
@@ -3,79 +3,69 @@
...
+  if (_.isString(key) && key.startsWith('$') && key in operatorsMap) {
+    return operatorsMap[key];
+  }
+  return key;
 };
 
 const buildCaslQuery = (ability, action, model) => {
   const query = rulesToQuery(ability, action, model, o => o.conditions);
-  return _.get(query, '$or[0].$and', {});
+  return _.get(query, '$or[0]', {});
```
Comment (alexandrebodin) [inline packages/core/admin/server/services/permission/permissions-manager/query-builers.js:31]: now that we support $or can we return the query directly ?

File: packages/core/admin/server/services/permission/permissions-manager/query-builers.js [hunk 5]
```diff
@@ -3,79 +3,69 @@
...
-      });
-    } else {
-      set(key, flattenDeep(value));
-    }
-  });
-
-  return result;
+  const transform = _.flow([unwrapDeep]);
+  const res = transform(caslQuery);
+  return res;
```
Comment (alexandrebodin) [inline packages/core/admin/server/services/permission/permissions-manager/query-builers.js:37]:
```suggestion
  return unwrapDeep(caslQuery);
```

File: packages/core/admin/server/services/permission/permissions-manager/query-builers.js [hunk 6]
```diff
@@ -3,79 +3,69 @@
...
+    (acc, v, k) => {
+      const key = mapKey(k);
+
+      if (_.isPlainObject(v)) {
+        if ('$elemMatch' in v) {
+          v = v.$elemMatch; // removing this key
+        }
+        _.setWith(acc, key, unwrapDeep(v));
+      } else if (_.isArray(v)) {
+        _.setWith(
```
Comment (alexandrebodin) [inline packages/core/admin/server/services/permission/permissions-manager/query-builers.js:59]: oneline ?

File: packages/core/utils/lib/convert-query-params.js [hunk 7]
```diff
@@ -206,21 +215,25 @@ const convertFieldsQueryParams = (fields, depth = 0) => {
 // NOTE: We could validate the parameters are on existing / non private attributes
 const convertFiltersQueryParams = filters => filters;
 
-// TODO: migrate
 const VALID_REST_OPERATORS = [
```
Comment (alexandrebodin) [inline packages/core/utils/lib/convert-query-params.js:218]: Do you still use them ? I think you can remove this constant now ?

**Review by alexandrebodin — CHANGES_REQUESTED** [review]
Looking good 🎉  a few comments

File: packages/core/admin/server/services/permission/permissions-manager/query-builers.js (hunk 4 above)
Comment (petersg83) [reply packages/core/admin/server/services/permission/permissions-manager/query-builers.js:31]: I guess so, lets try

File: packages/core/admin/server/services/permission/permissions-manager/query-builers.js (hunk 6 above)
Comment (petersg83) [reply packages/core/admin/server/services/permission/permissions-manager/query-builers.js:59]: it's prettier :'( Because of the arrow function I think

**Review by alexandrebodin — APPROVED** [review]

**Comment by alexandrebodin** [comment]
Nice job !

**Pull request merged**